[Bug]: `v2023.9.25` pip install fail on MacOS (in specific cases)

Hi! I noticed that the just-released pymatgen `v2023.9.25` [in this specific configuration cannot `pip install` correctly on **MacOS X86_64 GitHub runner**](https://github.com/amkrajewski/pqam-dparamhu2021/actions/runs/6302862819/job/17112125672) for Python 3.9 and Python 3.10, but can for Python 3.11.

**Notes:**
- Issue re-occurs when workflow is re-run.
- Issue does _not_ occur on Linux (Ubuntu).
- [Installs correctly under `v2023.9.10`](https://github.com/amkrajewski/pqam-dparamhu2021/actions/runs/6226021670/job/16897635425) on the same system.
- The only other dependency alongside it is `rpy2>3.5.0`. With `rpy2==3.5.14`, all dependency versions seem satisfied.
- It seems to be `numpy`-related. I attach a relevant snippet at the end. [Full log is here.](https://github.com/amkrajewski/pqam-dparamhu2021/actions/runs/6302862819/job/17112125672)
- When I tested 3 other packages that do not use `rpy2`, `v2023.9.25` installs as usual.

```
Compiling pymatgen/optimization/linear_assignment.pyx because it changed.
      [1/1] Cythonizing pymatgen/optimization/linear_assignment.pyx
      building 'pymatgen.optimization.linear_assignment' extension
      creating build/temp.macosx-11.7-x86_64-cpython-310
      creating build/temp.macosx-11.7-x86_64-cpython-310/pymatgen
      creating build/temp.macosx-11.7-x86_64-cpython-310/pymatgen/optimization
      gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -I/usr/local/opt/sqlite/include -I/usr/local/opt/sqlite/include -I/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-build-env-1za7ze_a/overlay/lib/python3.10/site-packages/numpy/core/include -I/Users/runner/hostedtoolcache/Python/3.10.13/x64/include/python3.10 -c pymatgen/optimization/linear_assignment.c -o build/temp.macosx-11.7-x86_64-cpython-310/pymatgen/optimization/linear_assignment.o
      <built-in>: error: unknown value ‘11.7’ of -mmacosx-version-min
      In file included from /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-build-env-1za7ze_a/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/ndarraytypes.h:1969,
                       from /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-build-env-1za7ze_a/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/ndarrayobject.h:12,
                       from /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-build-env-1za7ze_a/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/arrayobject.h:4,
                       from pymatgen/optimization/linear_assignment.c:1167:
      /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-build-env-1za7ze_a/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
       #warning "Using deprecated NumPy API, disable it with " \
        ^~~~~~~
      error: command '/usr/local/gfortran/bin/gcc' failed with exit code 1
      [end of output]
```

Have a lovely day! 
Adam

--- a/setup.py
+++ b/setup.py
@@ -34,7 +34,7 @@ setup(
         "matplotlib>=1.5",
         "monty>=3.0.2",
         "networkx>=2.2",
-        "numpy>=1.20.1",
+        "numpy>=1.25.0",
         "palettable>=3.1.1",
         "pandas",
         "plotly>=4.5.0",

